Realm/RLMResults.mm

```diff
@@ -364,6 +364,32 @@ - (RLMResults *)sortedResultsUsingDescriptors:(NSArray<RLMSortDescriptor *> *)pr
     });
 }
 
+- (RLMResults *)distinctResultsUsingKeyPaths:(NSArray<NSString *> *)keyPaths {
+    if (keyPaths.count == 0) {
+        @throw RLMException(@"Must specify at least one keypath.");
+    }
+    for (NSString *keyPath in keyPaths) {
+        if ([keyPath containsString:@"@"]) {
+            @throw RLMException(@"Cannot distinct on keypath '%@': KVC collection operators are not supported.", keyPath);
+        }
+    }
+    
+    return translateRLMResultsErrors([&] {
+        if (_results.get_mode() == Results::Mode::Empty) {
+            return self;
+        }
+        
+        __block std::vector<std::string> kp;
+        for (NSString *keyPath in keyPaths) {
+            kp.push_back(std::string([keyPath UTF8String]));
```

Comment:
The explicit conversion to `std::string` shouldn't be necessary here. `kp_push_back(keyPath.UTF8String)` should compile just fine.